packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -814,6 +817,26 @@ export const UnifiedDataTable = ({
     [visibleColumns, onSetColumns, shouldPrependTimeFieldColumn]
   );
 
+  // When columns are modified, reset the last column to auto width if only absolute
+  // width columns remain, to ensure the columns fill the available grid space
+  const prevColumns = useRef<string[]>();
+
+  useEffect(() => {
+    if (isEqual(prevColumns.current, visibleColumns)) {
+      return;
+    }
+
+    prevColumns.current = visibleColumns;
+
+    const hasAutoWidthColumn = visibleColumns.some(
+      (colId) => euiGridColumns.find((col) => col.id === colId)?.initialWidth == null
+    );
+
+    if (!hasAutoWidthColumn) {
+      onResize?.({ columnId: visibleColumns[visibleColumns.length - 1] });
+    }
+  }, [euiGridColumns, onResize, visibleColumns]);
+
```

Comment:
I wanted to save a search with all fixed width columns. But once saved/reloaded, the last column got reset and "Unsaved changes" badge appeared. Is it because of this code?

Reply:
Yeah, this turned out to be a pretty naive implementation. I tested it more thoroughly and ran into several edge cases where it would fail. I updated to a more robust approach here, which also allows consumers to have more control over the behaviour: https://github.com/elastic/kibana/pull/190288/commits/7b3583e1af1ebea49c99c0d105552e8d299d7509. I'll need to update tests tomorrow for the new approach.

Reply:
Tests updated: https://github.com/elastic/kibana/pull/190288/commits/4cf2b41ef9588a418a624874c4c57fd9195f8d2b